internal/ethapi/api.go

```diff
@@ -49,6 +49,11 @@ import (
 )
 
 var ErrHeaderNotFound = errors.New("header not found")
+var ErrHeaderNotFoundByHash = errors.New("header for hash not found")
```

Comment:
If both of the original errors (when emitted by geth) were wrapping `ethereum.NotFound` I think you could reduce it to a single `errors.Is(err, ethereum.NotFound)` error check and remove the error definitions and helper function.